code/game/objects/machinery/miner.dm

```diff
@@ -42,14 +42,19 @@
 	var/miner_integrity = 100
 	///Max health of the miner
 	var/max_miner_integrity = 100
-	///What type of upgrade it has installed , used to change the icon of the miner.
+	///What type of upgrade it has installed , used to change the behaviour of the miner.
 	var/miner_upgrade_type
+	///What type of miner this is. Options: default_miner, autominer, armor_miner, overclock_miner, platinum_miner
+	var/miner_type = "default_miner"
+	///If the miner is a platinum miner
+	var/isplatinum = FALSE
```

Comment:
This is extremely snowflakey

Reply:
yeah setup some flags or something instead of both of these vars, this is mega wew.